app-backend/batch/src/main/scala/com/azavea/rf/batch/ingest/model/SourceDefinition.scala

```diff
@@ -36,34 +36,37 @@ object SourceDefinition extends LazyLogging {
   @JsonCodec
   case class Overrides(
     uri: URI,
-    extent: Option[Extent],
-    extentCrs: Option[CRS],
-    crs: Option[CRS],
-    cellSize: Option[CellSize],
+    optionExtent: Option[Extent],
+    optionExtentCrs: Option[CRS],
+    optionCrs: Option[CRS],
+    optionCellSize: Option[CellSize],
     bandMaps: Array[BandMapping]
   ) {
     @SuppressWarnings(Array("OptionGet"))
     def toSourceDefinition: SourceDefinition = {
-      if (extent.isDefined && crs.isDefined && cellSize.isDefined && extentCrs.isDefined) {
-        SourceDefinition(
-          uri,
-          extent.get,
-          extentCrs.get,
-          crs.get,
-          cellSize.get,
-          bandMaps
-        )
-      } else {
-        logger.debug(s"Reading tiff tags: $uri")
-        lazy val tt = getTiffTags(uri)
-        SourceDefinition(
-          uri,
-          extent.getOrElse(tt.extent),
-          extentCrs.getOrElse(tt.crs),
-          crs.getOrElse(tt.crs),
-          cellSize.getOrElse(tt.cellSize),
-          bandMaps
-        )
+      (optionExtent, optionCrs, optionCellSize, optionExtentCrs) match {
+        case (Some(extent), Some(crs), Some(cellSize), Some(extentCrs)) => {
+          SourceDefinition(
+            uri,
+            extent,
+            extentCrs,
+            crs,
+            cellSize,
+            bandMaps
+          )
+        }
+        case _ => {
+          logger.debug(s"Reading tiff tags: $uri")
+          lazy val tt = getTiffTags(uri)
```

Comment:
Since there isn't a case in which we don't have to evaluate `getTiffTags`, why is this `lazy`? Especially now that we're not doing `getOrElse(tt.foo)`